mcan/src/messageram.rs

```diff
@@ -83,9 +80,13 @@ impl<C: Capacities> SharedMemory<C> {
     /// The peripheral uses 16-bit addressing for its memory configuration,
     /// offset from the start of system RAM. If `SharedMemory` is allocated
     /// outside the addressable region, it cannot be used.
-    pub fn is_addressable(&self) -> bool {
+    pub(crate) fn is_addressable(&self, eligible_message_ram_start: *const ()) -> bool {
+        // TODO: We should have a safety guarantee from
+        // `mcan_core::Dependencies::eligible_message_ram_start` implementor, so maybe
+        // this is too defensive
```

Comment:
I think it is reasonable to keep the mask, but if you really want to, I guess you could remove it. Either way, this comment should be changed to reference the specific part of the contract of `eligible_message_ram_start` that we rely on.

Reply:
I moved stuff around, you can take a look